### PR TITLE
Update Halo documentation for v0.3

### LINKED ISSUE #305: Update Halo module documentation
## Description
Update the halo documentation appearing on [readthedocs](https://skypy.readthedocs.io/en/latest/halo/index.html) to include plots and examples.

## References
See as an example [Pipeline](https://skypy.readthedocs.io/en/latest/pipeline/index.html)

### PR DESCRIPTION
**Pull request by Lucia-Fonseca**
## Description
This PR updates the halo documentation on [readthedocs](https://skypy.readthedocs.io/en/latest/halo/index.html) to add examples and plots. Merging this PR closes #305.
## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/master/CONTRIBUTING.md)
- [ ] Write unit tests
- [x] Write documentation strings
- [x] Assign someone from your working team to review this pull request
- [x] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

**Comment by Lucia-Fonseca** [comment]
I have followed @ntessore examples (e.g. pipeline) but I do not seem to render any plot. What am I missing?

File: docs/halo/examples/halo.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+cosmology: !astropy.cosmology.default_cosmology.get
```
Comment (ntessore) [inline docs/halo/examples/halo.yml:1]: this is done automatically now
```suggestion
```

File: docs/halo/index.rst [hunk 2]
```diff
@@ -4,6 +4,54 @@ Dark Matter Halos (`skypy.halo`)
 
 .. automodule:: skypy.halo
 
+You can reproduce figure 2 in Sheth and Tormen 1999
+and plot the collapse functions for different halo models:
+
+.. plot::
+
+    %matplotlib inline
```
Comment (ntessore) [inline docs/halo/index.rst:12]: not an IPython kernel

File: docs/halo/index.rst [hunk 3]
```diff
@@ -4,6 +4,54 @@ Dark Matter Halos (`skypy.halo`)
...
+    plt.ylabel(r'$f_c(\nu)$')
+    # show plot labels
+    plt.legend()
+
+You can also sample halos:
+
+.. plot::
+
+    halo_massST = pipeline['sheth-tormen']
+    halo_massPS = pipeline['press-schechter]
```
Comment (ntessore) [inline docs/halo/index.rst:43]:
```suggestion
    halo_massPS = pipeline['press-schechter']
```

**Review by ntessore — COMMENTED** [review]
The second plot directive is independent of the first, so you cannot reference the earlier results. The example should only produce a single plot with everything in it.

To catch these kinds of errors, it's best to run `make clean && make html` from your local `docs` folder.

File: docs/halo/examples/halo.yml (hunk 1 above)
Comment (Lucia-Fonseca) [reply docs/halo/examples/halo.yml:1]: oh true that! Thanks

**Comment by Lucia-Fonseca** [comment]
Still doesn't show the plot. Any idea why? Thanks!

**Comment by ntessore** [comment]
> Still doesn't show the plot. Any idea why? Thanks!

Your file is called 'halo.yml' not 'pipeline.yml'. Did you try to make the documentation locally?

**Comment by Lucia-Fonseca** [comment]
> > Still doesn't show the plot. Any idea why? Thanks!
> 
> Your file is called 'halo.yml' not 'pipeline.yml'. Did you try to make the documentation locally?

Oh,good spot! Locally works because I was using the right name of the config file

**Comment by Lucia-Fonseca** [comment]
Ready! Any idea of how to set the figure size correctly. It always looks small.

**Comment by ntessore** [comment]
> Ready! Any idea of how to set the figure size correctly. It always looks small.

I think it's ok as it is. If you want more legibility, perhaps 2 rows, 1 column might do the trick?

**Comment by Lucia-Fonseca** [comment]
Done

**Review by rrjbca — COMMENTED** [review]
I think it's important to show a simple skypy config file so that people learn how to write them e.g. [skypy.position](https://skypy.readthedocs.io/en/latest/position/index.html). Also your current config mixes together two calculations (collapse function and mass sampling) even though they are actually separate which could be confusing. I would suggest the following:
1) Show a simple script that calculates the three different collapse functions (no real need to do this in a config file)
2) Plot the collapse functions (but hide the script that does the plotting)
3) Show a config file for generating mass samples
4) Plot the mass samples (again hide the plotting script)

I would also increase the mass range and the number of samples so that the difference between the two distributions becomes clearer.

File: docs/halo/index.rst [hunk 4]
```diff
@@ -4,6 +4,94 @@ Dark Matter Halos (`skypy.halo`)
 
 .. automodule:: skypy.halo
 
+You can reproduce figure 2 in Sheth and Tormen 1999
+and plot the collapse functions for different halo models.
+
+.. code-block:: python
```
Comment (ntessore) [inline docs/halo/index.rst:10]:
```suggestion
.. plot::
   :include-source: true
   :nofigs: true
```

File: docs/halo/index.rst [hunk 5]
```diff
@@ -4,6 +4,94 @@ Dark Matter Halos (`skypy.halo`)
...
+    nu = np.square(delta_c / sigma)
+    params_model = (0.3, 0.7, 0.3, 1.686)
+
+    ST = sheth_tormen_collapse_function(sigma)
+    PS = press_schechter_collapse_function(sigma)
+    EM = ellipsoidal_collapse_function(sigma, params=params_model)
+
+
+.. plot::
+    :include-source: false
```
Comment (ntessore) [inline docs/halo/index.rst:37]:
```suggestion
   :include-source: false
   :context: close-figs
```

File: docs/halo/index.rst [hunk 6]
```diff
@@ -4,6 +4,94 @@ Dark Matter Halos (`skypy.halo`)
...
+
+    # Choose the halo models
+    ST = pipeline['sheth-tormen-collapse-function']
+    PS = pipeline['press-schechter-collapse-function']
+    EM = pipeline['ellipsoidal-collapse-function']
+
+    # plot different collapse functions
+    plt.loglog(pipeline['nu'], ST, label='Sheth-Tormen')
+    plt.loglog(pipeline['nu'], PS, label='Press-Schechter')
+    plt.loglog(pipeline['nu'], EM, label='Ellipsoidal')
```
Comment (ntessore) [inline docs/halo/index.rst:55]:
```suggestion
    # plot different collapse functions
    plt.loglog(nu, ST, label='Sheth-Tormen')
    plt.loglog(nu, PS, label='Press-Schechter')
    plt.loglog(nu, EM, label='Ellipsoidal')
```

**Review by ntessore — COMMENTED** [review]
@Lucia-Fonseca You can achieve Richard's points (1) and (2) using the `plot` directive without repeating yourself, by splitting the code up into a `:include-source: true`, `:nofigs: true` block that prepares the plot but doesn't plot anything, and a `:include-source: false` block with the same `:context:` that does the actual plotting.

For points (3) and (4) you can take a look at the positions example which shows a YAML file and then stealth-plots some results.

**Comment by Lucia-Fonseca** [comment]
Done. 
I do not show the config file because it quite long.

**Review by rrjbca — APPROVED** [review]

**Review by ntessore — APPROVED** [review]

**Pull request merged**